tests/xharness/TestReporter.cs

```diff
@@ -0,0 +1,490 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml;
+using System.Runtime.Serialization.Json;
+using Xharness.Execution;
+using Xharness.Listeners;
+using Xharness.Logging;
+using Xharness.Utilities;
+
+namespace Xharness {
+
+	public class TestReporterFactory : ITestReporterFactory {
+
+		public ITestReporter Create (IAppRunner appRunner, string device, ISimpleListener simpleListener, ILog log, ICrashSnapshotReporter crashReports)
+			=> new TestReporter (appRunner, device, simpleListener, log, crashReports);
+	}
+
+	// main class that gets the result of an executed test application, parses the results and provides information
+	// about the success or failure of the execution.
+	public class TestReporter : ITestReporter {
+
+		const string timeoutMessage = "Test run timed out after {0} minute(s).";
+		const string completionMessage = "Test run completed";
+		const string failureMessage = "Test run failed";
+
+		public TimeSpan Timeout { get; private set; }
+		public IProcessManager ProcessManager { get; set; } = new ProcessManager ();
+		public IResultParser ResultParser { get; set; } = new XmlResultParser ();
+		public ILog CallbackLog { get; private set; }
+
+		public bool? Success { get; private set; }
+		public Stopwatch TimeoutWatch { get; private set; }
+		public CancellationToken CancellationToken => cancellationTokenSource.Token;
+
+		public bool ResultsUseXml => runner.XmlJargon != XmlResultJargon.Missing;
+
+		readonly IAppRunner runner;
+		readonly ISimpleListener listener;
+		readonly ILogs crashLogs;
+		readonly ILog runLog;
+		readonly ICrashSnapshotReporter crashReporter;
+		readonly string deviceName;
+		bool waitedForExit = true;
+		bool launchFailure;
+		bool isSimulatorTest;
+		bool timedout;
+
+		readonly CancellationTokenSource cancellationTokenSource = new CancellationTokenSource ();
+
+		public TestReporter (IAppRunner appRunner, string device, ISimpleListener simpleListener, ILog log, ICrashSnapshotReporter crashReports)
```

Comment:
>IAppRunner appRunner [](start = 23, length = 20)

Let's not create dependency on AppRunner here as the TestReporter will go to the shared library. Just pass in the ILog and the RunMode or what is the reason you want this in.

Reply:
AppRunner contains a number of details that we need, not only the loggers. And there are diff logers that are used. The two classes are coupled but via an interface, so we can have a diff Runner implementation, there is not need to use the one we already have. Which I still think we can reuse.